ScreenManager.c

```diff
@@ -147,7 +147,7 @@ static void checkRecalculation(ScreenManager* this, double* oldTime, int* sortTi
       Machine_scan(host);
       if (!this->state->pauseUpdate)
          Machine_scanTables(host);
-      this->state->validUpdate = Platform_getValidState();
+      Platform_getFailedState(&this->state->failedUpdate);
```

Comment:
Is there a particular reason to use out parameters instead of a function return value?